### ISSUE TITLE
Update for Windows 10 "21H1" Build  19043

### ISSUE THREAD

**Issue by DavidGeeraerts** [ReneNyffenegger/patch-termsrv.dll#1]
Needs to be updated for the latest Windows 10 release "21H1" Build 19043
Any plans for doing so?

Should be able to extract the new hash string from [RDPWrapper](https://github.com/stascorp/rdpwrap/releases/tag/v1.6.2)

**Comment by DavidGeeraerts** [comment]
Latest config file associated with RDPWrapper.
Should contain the hex string?
[rdpwrap.ini.txt](https://github.com/ReneNyffenegger/patch-termsrv.dll/files/7308125/rdpwrap.ini.txt)

**Comment by ReneNyffenegger** [comment]
I'd certainly like to add the patch for the latest Windows release (and the missing ones, too). 
Correct me, if I am rwong, but it looks like the linked RDPWrapper has a different approach to using RDP than my PowerShell script: The RDPWrapper does not touch termsrv.dll while my script does so. Therefore, I believe that the rdpwrap.ini.txt file is of no help to me.

**Comment by DavidGeeraerts** [comment]
Correct, [RDPWrapper Project](https://github.com/stascorp/rdpwrap) doesn't touch [termsrv.dll](https://www.file.net/process/termsrv.dll.html). RDPWrapper creates an intermediary service between Service Control Manager and Terminal Services (a.k.a. Remote Desktop Service). This has the 'advantage' of not modifying the termsrv.dll file.
With Windows updates and/or Windows version release, the termsrv.dll file will get updated/change.
It becomes necessary to know what the 'new' HEX string is, and what to replace it with.
_(You know all this, I'm just writing it down ;-)_

For me, I prefer to go with changing the termsrv.dll file, and not install a service.
Seems lighter weight, and it saves me time having to go through the source code to ensure nothing questionable is going on.
Then there's: _"you should compile your own exe from source, after reviewing the source code."_

So much easier to just look at the PowerShell and decipher what is going on --more straightforward.

I thought you could use the latest rdpwrap.ini file to know what HEX string to look for, and what to replace it with --new HEX string.

How do you get the HEX string for each Windows 10 version: what to look for?
How do you get the HEX string for each Windows 10 version: what to replace with?

I hope I can help you get your Powershell script updated.

**Comment by infectedrusskiy** [comment]
@DavidGeeraerts 
> How do you get the HEX string for each Windows 10 version: what to look for? How do you get the HEX string for each Windows 10 version: what to replace with?

Did you ever come to an understanding about this?  I am on my own journey to best manage RDP and have hit the same roadblock.

**Comment by DavidGeeraerts** [comment]
@infectedrusskiy 
No, I stopped looking into it.
You can use Windows server RDP services for 180 days -- it's possible to reset the counter ;-)
